### PR TITLE
Document cluster.default_number_of_replicas and update index.number_of_replicas

### LINKED ISSUE #8525: [DOC] Document `cluster.default_number_of_replicas` and update `index.number_of_shards`
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.
`cluster.default_number_of_replicas` is an existing configuration but never documented. `index.number_of_shards` uses `cluster.default_number_of_replicas` as default value which is also not documented correctly. 

As two `index.number_of_shards` related bugs were fixed, we should have the correct documentation for the use of these two configurations.

https://github.com/opensearch-project/OpenSearch/issues/14810
https://github.com/opensearch-project/OpenSearch/issues/14783  

***Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
OS 2.18.
 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by chishui**
Document cluster.default_number_of_replicas and update index.number_of_replicas

### Description
#8525

### Issues Resolved
Closes #8525

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._
2.18

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 1]
```diff
@@ -79,6 +79,8 @@ OpenSearch supports the following dynamic cluster-level index settings:
 
 - `cluster.snapshot.shard.path.prefix` (String): Controls the fixed path prefix for snapshot shard-level blobs. This setting only applies when the repository `shard_path_type` setting is either `HASHED_PREFIX` or `HASHED_INFIX`. Default is an empty string, `""`.
 
+- `cluster.default_number_of_replicas` (Integer): Controls the default number of replicas for indexes. `index.number_of_replicas` will pick up this value if not configured. Default is 1.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:82]: "pick up" => "assume"? Should "1" be in code font?

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 2]
```diff
@@ -185,7 +187,7 @@ For more information about updating settings, including supported query paramete
 
 OpenSearch supports the following dynamic index-level index settings:
 
-- `index.number_of_replicas` (Integer): The number of replica shards each primary shard should have. For example, if you have 4 primary shards and set `index.number_of_replicas` to 3, the index has 12 replica shards. Default is 1.
+- `index.number_of_replicas` (Integer): The number of replica shards each primary shard should have. For example, if you have 4 primary shards and set `index.number_of_replicas` to 3, the index has 12 replica shards. It picks up `cluster.default_number_of_replicas` as its default value and if `cluster.default_number_of_replicas` is not set, default is 1.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:190]:
```suggestion
- `index.number_of_replicas` (Integer): The number of replica shards each primary shard should have. For example, if you have 4 primary shards and set `index.number_of_replicas` to 3, the index has 12 replica shards. It picks up `cluster.default_number_of_replicas` as its default value, and if `cluster.default_number_of_replicas` is not set, default is 1.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 3]
```diff
@@ -185,7 +187,7 @@ For more information about updating settings, including supported query paramete
...
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:191]: Above: "picks up" => "assumes"? Should "1" be in code font?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @chishui Please see my comments and changes and let me know if you have any questions. Thanks!

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @chishui!

**Pull request merged**